plugins/blip/basics/chartbasicsfactory.js

```diff
@@ -89,7 +92,47 @@ var BasicsChart = React.createClass({
       basicsData = _.assign({}, basicsData, _.cloneDeep(basicsState));
       var dataMunger = dataMungerMkr(this.props.bgClasses);
       dataMunger.reduceByDay(basicsData);
-      basicsData.data.reservoirChange.infusionSiteHistory = dataMunger.infusionSiteHistory(basicsData);
+
+      // get site changes from three event types
+      basicsData.data.cannulaPrime.infusionSiteHistory = dataMunger.infusionSiteHistory(basicsData, constants.SITE_CHANGE_CANNULA);
+      basicsData.data.tubingPrime.infusionSiteHistory = dataMunger.infusionSiteHistory(basicsData, constants.SITE_CHANGE_TUBING);
+      basicsData.data.reservoirChange.infusionSiteHistory = dataMunger.infusionSiteHistory(basicsData, constants.SITE_CHANGE_RESERVOIR);
+
+      var siteChangeEvents = [];
+
+      if (basicsData.data.tubingPrime.infusionSiteHistory.hasChangeHistory) {
+        siteChangeEvents.push(_.last(_.findLast(basicsData.data.tubingPrime.infusionSiteHistory, {type: constants.SITE_CHANGE}).data));
+      }
+      if (basicsData.data.cannulaPrime.infusionSiteHistory.hasChangeHistory) {
+        siteChangeEvents.push(_.last(_.findLast(basicsData.data.cannulaPrime.infusionSiteHistory, {type: constants.SITE_CHANGE}).data));
+      }
+      if (basicsData.data.reservoirChange.infusionSiteHistory.hasChangeHistory) {
+        siteChangeEvents.push(_.last(_.findLast(basicsData.data.reservoirChange.infusionSiteHistory, {type: constants.SITE_CHANGE}).data));
+      }
+
+      if (siteChangeEvents.length) {
+        // get latest pump upload:
+        var latestPump = _.last(_.sortBy(siteChangeEvents, 'time')).source;
```

Comment:
Yeah please ask for clarity on the "latest pump" language in the requirements on the Trello card. My (possibly imperfect) recollection is that it was supposed to be based on most recently uploaded pump... If my recollection turns out to be incorrect, please do rename the variable though, to something like `mostRecentSiteRelatedEvent` or something like that.